modal/_container_entrypoint.py

```diff
@@ -503,8 +503,11 @@ async def run_input(input_id, args, kwargs):
                     value = await res
                     await function_io_manager.enqueue_output.aio(input_id, started_at, output_index.value, value)
 
+        tasks = []
         async for input_id, args, kwargs in function_io_manager.run_inputs_outputs.aio(input_concurrency):
-            asyncio.create_task(run_input(input_id, args, kwargs))
+            tasks.append(asyncio.create_task(run_input(input_id, args, kwargs)))
+
+        await asyncio.gather(*tasks)
```

Comment:
Is this a reasonable way to clean up the stray tasks?

We would expect this `await` to complete exactly instantly. An alternative would be asserting `task.done() for task in tasks`.

Reply:
Yeah, asserting seems better. Also we should still use a `TaskContext` here, since it has a bunch of logic to clean up tasks when e.g. an exception happens.

Reply:
Yeah you're right. That makes a lot of sense.

Makes me wonder if we should hold off on the synchronous implementation until we make an equivalent `ThreadContext`?